packages/redux/src/raffles/reducer.ts

```diff
@@ -0,0 +1,156 @@
+  actionTypes,
+);
+const raffles = (state = INITIAL_STATE.raffles, action: AnyAction) => {
+  if (actionRafflesTypes.RAFFLE_ACTIONS.includes(action.type)) {
+    return {
+      ...state,
+      [action.meta.raffleId]: raffleReducer(
+        state[action.meta.raffleId],
+        action,
+      ),
+    };
+  }
+  return state;
+};
+
+const estimationReducer = createReducer(
+  'FETCH_RAFFLE_ESTIMATION',
+  { isLoading: false, error: null },
+  actionTypes,
+);
+const estimations = (state = INITIAL_STATE.estimations, action: AnyAction) => {
+  if (actionRafflesTypes.RAFFLE_ESTIMATION_ACTIONS.includes(action.type)) {
+    return {
+      ...state,
+      [action.meta.raffleId]: estimationReducer(
+        state[action.meta.raffleId],
+        action,
+      ),
+    };
+  }
+  return state;
+};
+
+const participationReducer = createReducer(
+  ['FETCH_RAFFLE_PARTICIPATION', 'UPDATE_RAFFLE_PARTICIPATION'],
+  { isLoading: false, error: null },
+  actionTypes,
+);
+const participations = (
+  state = INITIAL_STATE.participations,
+  action: AnyAction,
+) => {
+  if (actionRafflesTypes.RAFFLE_PARTICIPATION_ACTIONS.includes(action.type)) {
+    return {
+      ...state,
+      [action.meta.participationId]: participationReducer(
+        state[action.meta.participationId],
+        action,
+      ),
+    };
+  }
+  return state;
+};
+
+const participationCreationReducer = createReducer(
+  'CREATE_RAFFLE_PARTICIPATION',
+  { isLoading: false, error: null },
+  actionTypes,
+);
+const participationCreations = (
+  state = INITIAL_STATE.participationCreations,
+  action: AnyAction,
+) => {
+  if (
+    actionRafflesTypes.RAFFLE_PARTICIPATION_CREATION_ACTIONS.includes(
+      action.type,
+    )
+  ) {
+    return {
+      ...state,
+      [action.meta.raffleId]: participationCreationReducer(
+        state[action.meta.raffleId],
+        action,
+      ),
+    };
+  }
+  return state;
+};
+
+export const entitiesMapper = {
+  [actionTypes.RESET_RAFFLES_STATE]: (
+    state: NonNullable<StoreState['entities']>,
+  ) => {
+    const { raffles, raffleParticipations, raffleEstimations, ...rest } = state;
+    return rest;
+  },
+};
+
+export const getAllRaffles = (state: RafflesState) => state.allRaffles;
+export const getRaffles = (state: RafflesState) => state.raffles;
+export const getEstimations = (state: RafflesState) => state.estimations;
+export const getParticipations = (state: RafflesState) => state.participations;
+export const getParticipationCreations = (state: RafflesState) =>
+  state.participationCreations;
+
+const reducer = combineReducers({
+  allRaffles,
+  raffles,
+  participations,
+  participationCreations,
+  estimations,
+});
+
+/**
+ * Reducer for raffles state.
+ *
+ * @param state  - Current redux state.
+ * @param action - Action dispatched.
+ *
+ * @returns New state.
+ */
+const rafflesReducer: Reducer<T.RafflesState> = (state, action) => {
+  if (action.type === actionTypes.RESET_RAFFLES_STATE) {
```

Comment:
If this state is related to the specific user, this reducer should handle the LOGOUT_SUCCESS action as well. Take a look at bags reducer for an example. You will need to have an entitiesmapper for the `LOGOUT_SUCCESS` action as well to do the same as the reset raffles state action does to remove all raffles entities.

Reply:
done, pls double check

Reply:
Missing the entitiesMapper for the `LOGOUT_SUCCESS` take a look at bags reducer which contains an entities mapper for the logout success (and add a test for that case as well, the current test you have is inside the `entitiesMapper` describe block so you should test the entitiesMapper for the `LOGOUT_SUCCESS` action there and the test for the reducer when it receives a `LOGOUT_SUCCESS` action should be in another describe)

Reply:
added, pls double check